developer_support_script/check_build_stability.py

```diff
@@ -1,45 +1,162 @@
+"""
+A utility script to retrieve system tests from a jenkins job.
+
+The class currently acts as a singleton.
+"""
+
+# Dependencies
+import http.client as http
 import json
 import urllib.request
 from collections import Counter
 
-system_tests_url = "https://epics-jenkins.isis.rl.ac.uk/job/System_Tests/{}/testReport/api/json"
 
-test_metadata = "https://epics-jenkins.isis.rl.ac.uk/job/System_Tests/api/json"
+class SystemTestData:
+    """
+    Singleton class to host system test data pulled down from jenkins
+    """
+
+    def __init__(self, system_tests_url, test_metadata, test_return_quantity):
+        self.system_tests_url = system_tests_url
+        self.raw_test_metadata = test_metadata
```

Comment:
I don't have a particularily strong feeling on this, but Pylint consider this as too many attributes by default, if you wanted to remove some of them, some of these items are are used once, by a function also called in the init, they could be changed to just variables and passed into the function instead? I'll leave that up to you as I'm not sure either way is actually better, but it would make pylint happier.

Reply:
I think I'll leave this one for the time being as I'm not certain whether I might want some of these for the tabulate system tests issue i'm working on. Once I've finished that ticket, I'll create a flash review to change this should I find myself not pulling any of this out in favour of variables.